posthog/hogql_queries/insights/trends/trends_query_builder.py

```diff
@@ -706,7 +706,10 @@ def _events_filter(
                         elif property_clone["type"] == "person":
                             property_clone["key"] = f"events.person.properties.{property_clone['key']}"
                         property_clone["type"] = "data_warehouse"
-                    filters.append(property_to_expr(property_clone, self.team))
+                    expr = property_to_expr(property_clone, self.team)
+                    if property_clone["type"] in ("group", "element") and hasattr(expr, "left"):
+                        expr.left.chain = ["events", *expr.left.chain]
+                    filters.append(expr)
```

Comment:
This feels kind of magical, but I really don't know anything about what is expected when `is_data_warehouse_series` is true, and it's not too dissimilar to other code around it. I'd ping @Gilbert09 again just in case.

Reply:
Yeah I'm not so sure on this - looks like this was first introduced in https://github.com/PostHog/posthog/pull/27011. What happens if the data warehouse table has no `events` table attached? Should we not be testing for `modifiers.dataWarehouseEventsModifiers` here too?

Reply:
Good suggestion, updated with ec5aacabe3